jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/ForceObject.java

```diff
@@ -0,0 +1,87 @@
+package edu.uci.ics.jung.layout.spatial;
+
+import edu.uci.ics.jung.layout.model.Point;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** @author Tom Nelson */
+public class ForceObject<T> {
+
+  private static final Logger log = LoggerFactory.getLogger(ForceObject.class);
+  private static final double GRAVITY = 6.67e-11f;
+
+  /** location of p */
+  public final Point p;
+
+  /** force vector */
+  protected Point f;
+
+  /** mass */
+  protected double mass;
+
+  T element;
+
+  public ForceObject(T element, Point p, double mass) {
+    this.element = element;
+    this.p = p;
+    this.f = Point.ORIGIN;
+    this.mass = mass;
+  }
+
+  public ForceObject(T element, Point p) {
+    this(element, p, 1);
+  }
+
+  public ForceObject(T element, double x, double y) {
+    this(element, Point.of(x, y), 1000);
+  }
+
+  public ForceObject(T element, double x, double y, double mass) {
+    this.element = element;
+    p = Point.of(x, y);
+    this.mass = mass;
+  }
+
+  public Point getForce() {
+    return f;
+  }
+
+  private void addForceFrom(ForceObject other) {
+    double EPS = 3E4; // softening parameter
+    double dx = other.p.x - p.x;
+    double dy = other.p.y - p.y;
+    double dist = Math.sqrt(dx * dx + dy * dy);
+    double force = (mass * other.mass * GRAVITY) / (dist * dist + EPS * EPS);
```

Comment:
if you want to optimize this a tiny bit, you can store the squared distance and only take the root when needed below, to avoid recalculating the squared difference.